src/memory/attribute.rs

```diff
@@ -731,54 +1050,45 @@ impl<'topology> MemoryAttribute<'topology> {
     /// [`NeedInitiator`]: InitiatorInputError::NeedInitiator
     /// [`UnwantedInitiator`]: InitiatorInputError::UnwantedInitiator
     #[doc(alias = "hwloc_memattr_get_value")]
-    pub fn value<'initiator>(
+    pub fn value(
         &self,
-        initiator: Option<impl Into<MemoryAttributeLocation<'initiator>>>,
+        initiator: Option<MemoryAttributeLocation<'_>>,
```

Comment:
`Option<impl Trait>` is a bad idea because `None` trivially fails type inference.

By removing it here and below, we're also able to get rid of the cognitive load of polymorphization.